File: performance-noop/src/main/java/io/github/xanderwang/performance/PERF.java
package io.github.xanderwang.performance;

import java.io.File;

public class PERF {

  public static class Builder {

    public Builder checkUI(boolean check) {
      return this;
    }

    public Builder checkUI(boolean check, long blockTime) {
      return this;
    }

    public Builder checkThread(boolean check) {
      return this;
    }

    public Builder checkThread(boolean check, long threadBlockTime) {
      return this;
    }

    public Builder checkFps(boolean check) {
      return this;
    }

    public Builder checkFps(boolean check, long fpsIntervalTime) {
      return this;
    }

    public Builder checkIPC(boolean check) {
      return this;
    }

    public Builder checkBitmap(boolean check) {
      return this;
    }

    public Builder globalTag(String tag) {
      return this;
    }

    public Builder cacheDirSupplier(IssueSupplier<File> cache) {
      return this;
    }

    public Builder maxCacheSizeSupplier(IssueSupplier<Integer> cacheSize) {
      return this;
    }

    public Builder uploaderSupplier(IssueSupplier<LogFileUploader> uploader) {
      return this;
    }

    public Builder logLevel(int level) {
      return this;
    }

    public Builder build() {
      return this;
    }

  }

  public interface IssueSupplier<T> {
    T get();
  }

  public interface LogFileUploader {
    boolean upload(File logFile);
  }

  public static void init(Builder builder) {

  }

}

File: performance-noop/src/main/java/io/github/xanderwang/performance/PERF.kt
package io.github.xanderwang.performance

import java.io.File

object PERF {

    fun init(builder: Builder?) {
    }

    class Builder {
        fun checkUI(check: Boolean): Builder {
            return this
        }

        fun checkUI(check: Boolean, blockTime: Long): Builder {
            return this
        }

        fun checkThread(check: Boolean): Builder {
            return this
        }

        fun checkThread(check: Boolean, threadBlockTime: Long): Builder {
            return this
        }

        fun checkFps(check: Boolean): Builder {
            return this
        }

        fun checkFps(check: Boolean, fpsIntervalTime: Long): Builder {
            return this
        }

        fun checkIPC(check: Boolean): Builder {
            return this
        }

        fun checkIPC(check: Boolean, ipcBlockTime: Long): Builder {
            return this
        }

        fun checkBitmap(check: Boolean): Builder {
            return this
        }

        fun filterClass(filter: Boolean): Builder {
            return this
        }

        fun globalTag(tag: String?): Builder {
            return this
        }

        fun cacheDirSupplier(cache: IssueSupplier<File?>?): Builder {
            return this
        }

        fun maxCacheSizeSupplier(cacheSize: IssueSupplier<Int?>?): Builder {
            return this
        }

        fun uploaderSupplier(uploader: IssueSupplier<LogFileUploader?>?): Builder {
            return this
        }

        fun logLevel(level: Int): Builder {
            return this
        }

        fun build(): Builder {
            return this
        }
    }

    interface IssueSupplier<T> {
        fun get(): T
    }

    interface LogFileUploader {
        fun upload(logFile: File?): Boolean
    }
}